app/components/primer/clipboard_copy.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use ClipboardCopy to populate copy values from the DOM
+  class ClipboardCopy < Primer::Component
+    status :alpha
+
+    # Optional target element that holds the data the user will copy.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :target, lambda { |**system_arguments|
```

Comment:
Looking around dotcom, I think it might make sense to also have an `icon` slot that wraps an OcticonComponent, as we often use an octicon with this component. 

This can of course wait for another PR if you'd prefer ❤️

Reply:
Taking this one step further, we almost _always_ use this with the `clippy` icon, but inconsistently swap it for the `check` icon when clicked. Perhaps we could encode that mode as an option on the component?

Reply:
I think including the icon here makes sense. I think defaulting to the icon makes sense, and we can let users pass `icon: nil` to render the component without the icon.